lib/kitchen/driver/ec2.rb

```diff
@@ -354,6 +361,8 @@ def hostname(server, interface_type = nil)
           potential_hostname = nil
           INTERFACE_TYPES.values.each do |type|
             potential_hostname ||= server.send(type)
+            # AWS returns an empty string if the dns name isn't populated yet
```

Comment:
Why not default to private_ip unless specified? 

It seems like it would be the most likely to be available, followed by public_ip and then DNS. Looks like there could be reduced complexity unless there is something I am missing. 

What is the behavior of not defining the hostname? (returning nil)